scripts/e2e-custom.sh

```diff
@@ -71,13 +72,26 @@ function init_e2e_tests() {
     DATA_HOME=$(python -c 'import platformdirs; print(platformdirs.user_data_dir())')
     CACHE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_cache_dir())')
     STATE_HOME=$(python -c 'import platformdirs; print(platformdirs.user_state_dir())')
+
     # ensure that our mock e2e dirs exist
     for dir in "${CONFIG_HOME}" "${DATA_HOME}" "${STATE_HOME}" "${CACHE_HOME}"; do
     	mkdir -p "${dir}"
     done
 
     E2E_LOG_DIR="${HOME}/log"
     mkdir -p "${E2E_LOG_DIR}"
+
+    # set appropriate feature gating flag
+    if [ "$DEV_PREVIEW_FEATURE_GATING" -eq 1 ]; then
```

Comment:
Note: you probably have these env vars in your shell, so you're not seeing the `DEV_PREVIEW_FEATURE_GATING: unbound variable` error.
Since we have `set -u` at the beginning, we have to ensure that the env variables are defined.
We can use either `-n` or provide a default as in:
```
    if [ ${DEV_PREVIEW_FEATURE_GATING:-} -eq 1 ]; then
```

Explanation of `set -u`:
```
Treat unset variables and parameters other than the special parameters ‘@’ or ‘*’, or array variables subscripted with ‘@’ or ‘*’, as an error when performing parameter expansion. An error message will be written to the standard error, and a non-interactive shell will exit.
```